### PR TITLE
Feat(eos_validate_state): skip lldp topology for shutdown

### LINKED ISSUE #2110: `eos_validate_state` lldp_topology check skip shutdown interfaces
### Enhancement summary

At the moment, using `is_deployed: false` in the inventory and `shutdown_interfaces_towards_undeployed_peers: true` will prevent false positives by eos_validate_state for **interfaces_state** checks but **lldp_topology** will still raise failures for these peers that are not yet deployed.

### Which component of AVD is impacted

eos_validate_state

### Use case example

If an inventory has some devices marked as  `is_deployed: false`, then eos_validate_state should not generate **lldp_topology** failures for the interfaces towards these missing devices.

### Describe the solution you would like

Marking a device as `is_deployed: false`  and `shutdown_interfaces_towards_undeployed_peers: true` would cause the **lldp_topology** check to skip the undeployed device.

I believe the logic here could be improved to check if the interface is shutdown : https://github.com/aristanetworks/ansible-avd/blob/8abbc4dcfe28f7ed2d35a8471e5fce4d0902cd0d/ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_no_fqdn.yml#L25

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by chetryan**
## Change Summary

Skip checking LLDP topology for interfaces that are `shutdown: true`.

When a device in the inventory has `is_deployed: false` and the global value `shutdown_interfaces_towards_undeployed_peers: true` is set, the interfaces towards devices that have not been deployed will be in the shutdown state. 
In the above case, we should skip checking for lldp neighborships as we will not find any lldp peers if the interface is shutdown.

## Related Issue(s)

Fixes #2110

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes

Adds a check to make sure interface is not shutdown before checking lldp neighborship. 
`not (ethernet_interface.value.shutdown is undefined or ethernet_interface.value.shutdown) and` so that the logic becomes: 

```
when: |
    (ethernet_interfaces is defined and ethernet_interfaces is not none) and
    not (ethernet_interface.value.shutdown is undefined or ethernet_interface.value.shutdown) and
    (ethernet_interface.value.peer is defined and ethernet_interface.value.peer is not none) and
    (ethernet_interface.value.peer_interface is defined and ethernet_interface.value.peer_interface is not none) and
    (hostvars[ethernet_interface.value.peer] is defined and hostvars[ethernet_interface.value.peer] is not none) and
    (hostvars[ethernet_interface.value.peer]['dns_domain'] is defined and hostvars[ethernet_interface.value.peer]['dns_domain'] is not none)
```

## How to test

_There are currently no Molecule tests for `eos_validate_state`_

### User Checklist

- [x] Tested on actual topology inside Lab. 

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
For the case of `shutdown` being `undefined` I don't think we should skip this test. EOS has global knobs that can inverse the default state of the interface, so you cannot deduct it the interface is shutdown or not based on a missing variable. Apart from that, I support the rest of the change. I did some fundamental changes to validate_state in another PR, so we will probably need some rebasing and fix merge conflicts when the first one merges.

The logic could be `... and ethernet_interface.shutdown is not arista.avd.defined(true) and ...`

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_fqdn.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@
     loop_var: ethernet_interface
   when: |
     (ethernet_interfaces is defined and ethernet_interfaces is not none) and
+    not (ethernet_interface.value.shutdown is undefined or ethernet_interface.value.shutdown) and
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_fqdn.yml]:
```suggestion
    (ethernet_interface.shutdown is not arista.avd.defined(true)) and
```

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_no_fqdn.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@
     loop_var: ethernet_interface
   when: |
     (ethernet_interfaces is defined and ethernet_interfaces is not none) and
+    not (ethernet_interface.value.shutdown is undefined or ethernet_interface.value.shutdown) and
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_no_fqdn.yml]:
```suggestion
   (ethernet_interface.shutdown is not arista.avd.defined(true)) and
```

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_fqdn.yml (hunk 1 above)
Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_fqdn.yml]: Same suggestion as recommended by Claus.

**Review by ClausHolbechArista — APPROVED** [review]

**Comment by ClausHolbechArista** [comment]
Approved but please make tests pass
https://github.com/aristanetworks/ansible-avd/actions/runs/3386366101/jobs/5625683269

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_no_fqdn.yml [hunk 3]
```diff
@@ -23,6 +23,7 @@
     loop_var: ethernet_interface
   when: |
     (ethernet_interfaces is defined and ethernet_interfaces is not none) and
+   (ethernet_interface.shutdown is not arista.avd.defined(true)) and
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_no_fqdn.yml]:
```suggestion
    (ethernet_interface.value.shutdown is not arista.avd.defined(true)) and
```

Sorry, I provided wrong suggestion!

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_fqdn.yml [hunk 4]
```diff
@@ -23,6 +23,7 @@
     loop_var: ethernet_interface
   when: |
     (ethernet_interfaces is defined and ethernet_interfaces is not none) and
+    (ethernet_interface.shutdown is not arista.avd.defined(true)) and
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_fqdn.yml]:
```suggestion
    (ethernet_interface.value.shutdown is not arista.avd.defined(true)) and
```

File: ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_no_fqdn.yml (hunk 3 above)
Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_validate_state/tasks/lldp_topology_no_fqdn.yml]: As well as the wrong indentation!

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**